core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1174,7 +1173,7 @@ protected Scope visitLateral(Lateral node, Optional<Scope> scope)
                 return Optional.empty();
             }
 
-            String storageTable = String.valueOf(optionalView.get().getProperties().getOrDefault(STORAGE_TABLE, ""));
+            String storageTable = optionalView.get().getStorageTable();
```

Comment:
It doesn't look like we're guaranteeing that if `properties` has `storage_table` then `storageTable` is also populated in `ConnectorMaterializedViewDefinition`
CreateMaterializedViewTask#execute creates a ConnectorMaterializedViewDefinition where storageTable is null but  properties might be populated.

Reply:
> It doesn't look like we're guaranteeing that if properties has storage_table then storageTable is also populated in ConnectorMaterializedViewDefinition

This is a bug IMO. We should always look at `getStorageTable()` instead of `STORAGE_TABLE` property

> CreateMaterializedViewTask#execute creates a ConnectorMaterializedViewDefinition where storageTable is null but properties might be populated.

This is because engine doesn't know storage table name when creating definition